### PR TITLE
 Fix ballerina upgrade

### PR DESCRIPTION
**Pull request by hasuniea**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: components/micro-gateway-core/src/main/ballerina/gateway/utils/validation_util.bal [hunk 1]
```diff
@@ -395,7 +393,7 @@ public function merge(json target, json source) returns (json) {
         } else {
             //if there is a value available for this property in the source model, check whether there are any
             //exceptional keys inside that property
-            if (sourceProperty.getKeys() != null) {
+            if (sourceProperty.getKeys().length() > 0) {
```
Comment (Rajith90) [inline components/micro-gateway-core/src/main/ballerina/gateway/utils/validation_util.bal]: What if  sourceProperty.getKeys() is null, then calling the length() of a null will be a issue I guess

File: components/micro-gateway-core/src/main/ballerina/gateway/utils/validation_util.bal [hunk 2]
```diff
@@ -465,7 +463,7 @@ public function validateType(string name, json value, json field, json models) r
     Result result = new();
     //getting type defined in the model/field
     string expectedType = field[TYPE].toString();
-    if (expectedType == null) {//if the field doesn't have a type it may have a reference to another model
+    if (expectedType.length() > 0) {//if the field doesn't have a type it may have a reference to another model
```
Comment (Rajith90) [inline components/micro-gateway-core/src/main/ballerina/gateway/utils/validation_util.bal]: expectedType can be null, so calling lenght() might be an issue

File: components/micro-gateway-core/src/main/ballerina/gateway/utils/validation_util.bal [hunk 3]
```diff
@@ -431,12 +429,12 @@ public function createReturnObject(error[]|error err, string modelName) returns
...
             int index = 0;
             foreach var item in err {
-                result.getErrorMessages[index] = item.message;
+                result.getErrorMessages[index] = item.reason();
                 index = index +1;
             }
         }
     }
-    if (modelName != null) {
+    if (modelName.length() > 0) {
```
Comment (Rajith90) [inline components/micro-gateway-core/src/main/ballerina/gateway/utils/validation_util.bal]: modelName can be null

File: components/micro-gateway-core/src/main/ballerina/gateway/utils/validation_util.bal [hunk 4]
```diff
@@ -871,10 +872,10 @@ public function validateProperties(json target, json model, json models) returns
     //get all details in referenced models and merge them into a one model
     var referenceModel = dereferenceModel(target, model, models);
     //if the properties provided in the target object is not defined in the model, send an error
-    if (targetKeys != null) {
+    if (targetKeys.length() > 0) {
```
Comment (Rajith90) [inline components/micro-gateway-core/src/main/ballerina/gateway/utils/validation_util.bal]: What if targteKeys become null.

File: components/micro-gateway-core/src/main/ballerina/gateway/utils/validation_util.bal (hunk 3 above)
Comment (hasuniea) [reply components/micro-gateway-core/src/main/ballerina/gateway/utils/validation_util.bal]: ModelName is a string value.

**Pull request merged**